includes/Services/Cloudflare.php

```diff
@@ -73,6 +73,19 @@ private function makeRequest( array $urls ): array {
 			sprintf( 'Added %d files to Cloudflare request: %s', count( $urls ), json_encode( $urls, JSON_THROW_ON_ERROR ) )
 		);
 
+		$postData = [ 'files' => $urls ];
+
+		if ( $this->extensionConfig->get( 'MultiPurgeCloudFlareCacheByDeviceType' ) ) {
+			foreach ( $urls as $url ) {
+				$postData['files'][] = [
+					'url' => $url,
+					'headers' => [ 'CF-Device-Type' => 'mobile' ]
+				];
+			}
+		}
```

Comment:
While this does work this needs to be moved up to the `getPurgeRequest` function due to CloudFlare's 30 url purge limit:
> For Zones on Free/Pro/Business plan, you may purge up to 30 URLs in one API call.

https://developers.cloudflare.com/cache/how-to/purge-cache/purge-by-single-file/

This change introduces the possibility for a purge request to fail, if a chunk contains more than 15 urls.

You could alternatively change the number of chunks in [L49](https://github.com/octfx/mediawiki-extensions-MultiPurge/blob/6eb6ad3d9f1477dc6104fc884c0ccf5ff8f5fc01/includes/Services/Cloudflare.php#L49) based on the value of `$wgMultiPurgeCloudFlareCacheByDeviceType`.

Reply:
Ah, thanks! I hadn't entirely realized that. I've gone with your suggestion.